awx/main/scheduler/task_manager.py

```diff
@@ -376,45 +374,71 @@ def should_update_related_project(self, job, latest_project_update):
             return True
         return False
 
+    def gen_dep_for_job(self, task):
+        created_dependencies = []
+        dependencies = []
+        # TODO: Can remove task.project None check after scan-job-default-playbook is removed
+        if task.project is not None and task.project.scm_update_on_launch is True:
+            latest_project_update = self.get_latest_project_update(task.project_id)
+            if self.should_update_related_project(task, latest_project_update):
+                project_task = self.create_project_update(task)
+                created_dependencies.append(project_task)
+                dependencies.append(project_task)
+            else:
+                dependencies.append(latest_project_update)
+
+        # Inventory created 2 seconds behind job
+        try:
+            start_args = json.loads(decrypt_field(task, field_name="start_args"))
+        except ValueError:
+            start_args = dict()
+        # generator for inventory sources related to this task
+        task_inv_sources = (invsrc for invsrc in self.all_inventory_sources if invsrc.inventory == task.inventory)
```

Comment:
Especially when looping over a global jobs list like `self.all_inventory_sources`, I would like to stick with `*_id` fields for performance reasons (even if it may or may not make a difference). So here, I would make the conditional be `if invsrc.inventory_id == task.inventory_id`

Reply:
well I get that you're just moving the code

Reply:
made the change